packages/js/product-editor/src/hooks/use-product-template/test/use-product-template.test.ts

```diff
@@ -70,49 +91,63 @@ describe( 'useProductTemplate', () => {
 
 	it( 'should return the product template if it exists', () => {
 		const { result } = renderHook( () =>
-			useProductTemplate( 'template-3', 'simple' )
+			useProductTemplate( 'template-3', 'simple', 'product' )
 		);
 
 		expect( result.current.productTemplate?.id ).toEqual( 'template-3' );
 	} );
 
 	it( 'should return the first product template with a matching type in the productData if no matching product template by id', () => {
 		const { result } = renderHook( () =>
-			useProductTemplate( 'invalid-template-id', 'grouped' )
+			useProductTemplate( 'invalid-template-id', 'grouped', 'product' )
 		);
 
 		expect( result.current.productTemplate?.id ).toEqual( 'template-2' );
 	} );
 
 	it( 'should return the first product template with a matching type in the productData if no product template id is set', () => {
 		const { result } = renderHook( () =>
-			useProductTemplate( undefined, 'grouped' )
+			useProductTemplate( undefined, 'grouped', 'product' )
 		);
 
 		expect( result.current.productTemplate?.id ).toEqual( 'template-2' );
 	} );
 
+	it( 'should return the first product template with a matching post type when no product template id or product type is set', () => {
```

Comment:
Really appreciate how you always add so many tests.  Nicely done, Matt!

Reply:
Thanks! The tests are key for APIs like this.